### PR TITLE
Replace no_std feature with std feature

### LINKED ISSUE #23: Flip no_std flag
Features should be additive.

### LINKED ISSUE #47: serde requires std
Enabling the _serde_ feature pulls in std via a serde default feature. The serdes std feature is not needed here, is it?

Related to #23

### PR DESCRIPTION
**Pull request by jhbruhn**
This PR fixes #23 and fixes #47 by replacing the no_std flag with an additive std flag and disabling default features on serde. 

libm is not an optional dependency anymore because it's functions are required in a no_std context. But as they are only used in said context, it should get optimized away if the std flag is enabled.
I'm happy to change that though if anyone has a better suggestion.

I have chosen not to explicitly enable serdes std flag as it is not required in this library and any consuming application will have that enable anyways if they are using serde in an std context, or am I missing a case there?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,8 +89,8 @@ fn main() {
 
 The crate contains few features to disable or enable certain functionalities:
 
-* no_std
-    * Removes functionality that Rust std library provides
+* std
```
Comment (eldruin) [inline README.md:92]:
```suggestion
* `std`
```

File: README.md [hunk 2]
```diff
@@ -89,8 +89,8 @@ fn main() {
...
+    * Enables functionality that Rust std library provides instead of using libm for some maths functions
```
Comment (eldruin) [inline README.md:93]:
```suggestion
    * Enables functionality that Rust standard library provides instead of using `libm` for some math functions
```

**Review by eldruin — COMMENTED** [review]
Looks good so far. Could you:
- Adapt the CI `.github/workflows/build.yml` changing the `no_std` feature to `std?
- Add an entry to the changelog
- See if it is possible to use the `std` for the calculations where `libm` is currently explicitly used and only use the `libm` version in the `no_std` case? (e.g [here](https://github.com/rust-embedded-community/rust-measurements/blob/043968f73dfd60ea219ec48a0580084464adccc0/src/humidity.rs#L88))

**Comment by jhbruhn** [comment]
Unfortunately I didn't understand your last point about libm, could you rephrase that? Are you suggesting to use libm-based calculations both for std and no_std targets? 
Currently in this PR, no_std targets are using libm implementations while std targets are using the std-based implementations.
IMO, we could just use the libm implementations without loosing anything, but maybe I'm missing important details about libm here?

**Comment by eldruin** [comment]
Sorry about that. I mean it would be good to avoid the `libm` dependency completely if including the `std`. At the moment `libm` seems to be used only for a few calculations in the `humidity.rs` file.
For example, one could use either `libm::log` for `no-std` or `f64::ln` with `std` enabled.

**Comment by jhbruhn** [comment]
I agree, but I think this is not possible right now (https://github.com/rust-lang/cargo/issues/1839) without introducing a new `no-std` feature, which would again mean that we could activate both `std` and `no-std` with is...semantically interesting.

**Comment by eldruin** [comment]
Yeah that is a limitation. We will have to compile it no matter what. However, we could simply not use it and use the `std` function versions instead if enabled, as this may mean using more optimized math function implementations.

**Comment by jhbruhn** [comment]
Agreed, that is already the case in this PR. The only place I could find that uses libm is `humidity.rs`, which has feature-gated implementations for some fns depending on whether `std` is enabled or not.

**Comment by eldruin** [comment]
Interesting. Sorry about that. I saw now that the whole functions are simply not available when `std` is enabled (same situation as before). However, it does not make much sense to me how this has been the whole time. Maybe it was only an artifact of the `no_std` feature situation.
Could you try enabling them when `std` is enabled as well (but now using the `std` math functions)?

**Comment by jhbruhn** [comment]
Which ones are you talking about? `humidity.rs` provides alternative implementations for some functions, if `std` is enabled, std is used, if `std` is not enabled, `libm` is used.
`angle.rs` seems to have some functions which are only enabled with the `std` features, I can add additional libm implementations for those.

**Review by eldruin — APPROVED** [review]
Uh, sorry for my confusion and thanks for bearing with me.
This PR is of course great as it is.
We can enable the trigonometric operations on `Angle` when on `no-std` with `libm` on a separate PR if you are interested.
bors r+
